npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -122,20 +123,63 @@ func (dp *DataPlane) updatePod(pod *updateNPMPod) error {
 	if !ok {
 		// ignore this err and pod endpoint will be deleted in ApplyDP
 		// if the endpoint is not found, it means the pod is not part of this node or pod got deleted.
-		klog.Warningf("[DataPlane] did not find endpoint with IPaddress %s", pod.PodIP)
+		klog.Warningf("[DataPlane] did not find endpoint with IPaddress %s for pod %s", pod.PodIP, pod.PodKey)
 		return nil
 	}
 
-	if endpoint.podKey == unspecifiedPodKey {
-		// while refreshing pod endpoints, newly discovered endpoints are given an unspecified pod key
-		if endpoint.isStalePodKey(pod.PodKey) {
-			// NOTE: if a pod restarts and takes up its previous IP, then its endpoint would be new and this branch would be taken.
-			// Updates to this pod would not occur. Pod IPs are expected to change on restart though.
-			// See: https://stackoverflow.com/questions/52362514/when-will-the-kubernetes-pod-ip-change
-			// If a pod does restart and take up its previous IP, then the pod can be deleted/restarted to mitigate this problem.
-			klog.Infof("ignoring pod update since pod with key %s is stale and likely was deleted", pod.PodKey)
+	// While refreshing pod endpoints, newly discovered endpoints are given an unspecified pod key.
+	// Additionally, a pod key may be stale if the pod was wrongly assigned to the endpoint for this scenario:
+	// 1. pod A previously had IP i and EP x
+	// 2. pod A restarts w/ no ip AND NPM restarts AND pod B comes up with the same IP i and EP y
+	// 3. controller processes an update event for pod A with IP i before the update event for pod B with IP i, so pod A is wrongly assigned to EP y
+	if endpoint.isStalePodKey(pod.PodKey) {
```

Comment:
ok to move this up here out of the if-unspecified-pod-key condition because it's impossible to have a podKey be assigned to the endpoint and be stale (endpoints only touched in this function and refreshPodEndpoints)